sci-geosciences/qgis/qgis-2.18.2-r3.ebuild

```diff
@@ -0,0 +1,198 @@
+		dev-python/python-dateutil[${PYTHON_USEDEP}]
+		dev-python/httplib2[${PYTHON_USEDEP}]
+		dev-python/jinja[${PYTHON_USEDEP}]
+		dev-python/markupsafe[${PYTHON_USEDEP}]
+		dev-python/pygments[${PYTHON_USEDEP}]
+		dev-python/pytz[${PYTHON_USEDEP}]
+		dev-python/requests[${PYTHON_USEDEP}]
+		dev-python/six[${PYTHON_USEDEP}]
+		postgres? ( dev-python/psycopg:2[${PYTHON_USEDEP}] )
+		${PYTHON_DEPS}
+	)
+	dev-db/sqlite:3
+	>=dev-db/spatialite-4.1.0
+	app-crypt/qca:2[qt4,ssl]
+"
+
+DEPEND="${RDEPEND}
+	sys-devel/bison
+	sys-devel/flex"
+
+DOCS=( BUGS ChangeLog NEWS )
+
+PATCHES=( "${FILESDIR}"/${P}-desktop-icon.patch )
+
+# Disabling test suite because upstream disallow running from install path
+RESTRICT="test"
+
+pkg_setup() {
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+
+	cd src/plugins || die
+	use georeferencer || cmake_comment_add_subdirectory "georeferencer"
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DQGIS_MANUAL_SUBDIR=/share/man/
+		-DBUILD_SHARED_LIBS=ON
+		-DQGIS_LIB_SUBDIR=$(get_libdir)
+		-DQGIS_PLUGIN_SUBDIR=$(get_libdir)/qgis
+		-DWITH_INTERNAL_DATEUTIL=OFF
+		-DWITH_INTERNAL_HTTPLIB2=OFF
+		-DWITH_INTERNAL_JINJA2=OFF
+		-DWITH_INTERNAL_MARKUPSAFE=OFF
+		-DWITH_INTERNAL_PYGMENTS=OFF
+		-DWITH_INTERNAL_PYTZ=OFF
+		-DWITH_INTERNAL_QWTPOLAR=OFF
+		-DWITH_INTERNAL_SIX=OFF
+		-DPEDANTIC=OFF
+		-DWITH_APIDOC=OFF
+		-DWITH_QSPATIALITE=ON
+		-DENABLE_TESTS=OFF
+		-DWITH_BINDINGS="$(usex python)"
+		-DWITH_GRASS7="$(usex grass)"
+		$(usex grass "-DGRASS_PREFIX=/usr/" "")
+		-DWITH_ORACLE="$(usex oracle)"
+		-DWITH_POSTGRESQL="$(usex postgres)"
+		-DWITH_PYSPATIALITE="$(usex python)"
+		-DWITH_SERVER="$(usex mapserver)"
+	)
+
+	if has_version '>=x11-libs/qwtpolar-1' &&  has_version 'x11-libs/qwt:5' ; then
+		elog "Both >=x11-libs/qwtpolar-1 and x11-libs/qwt:5 installed. Force build with qwt6"
+		if has_version '>=x11-libs/qwt-6.1.2' ; then
+			mycmakeargs+=(
+				"-DQWT_INCLUDE_DIR=/usr/include/qwt6"
+				"-DQWT_LIBRARY=/usr/$(get_libdir)/libqwt6-qt4.so"
+			)
+		else
+			mycmakeargs+=(
+				"-DQWT_INCLUDE_DIR=/usr/include/qwt6"
+				"-DQWT_LIBRARY=/usr/$(get_libdir)/libqwt6.so"
+			)
+		fi
+	fi
+
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	domenu debian/qgis.desktop
+
+	local types="qgs qml qlr qpt"
+	local sizes="16 22 24 32 48 96 128 256"
+	for size in ${sizes}; do
+		newicon -s ${size} debian/${PN}-icon${size}x${size}.png ${PN}.png
+		newicon -c mimetypes -s ${size} debian/${PN}-mime-icon${size}x${size}.png ${PN}-mime.png
+		for type in ${types}; do
+			newicon -c mimetypes -s ${size} debian/${PN}-${type}${size}x${size}.png ${PN}-${type}.png
+		done
+	done
+	newicon -s scalable images/icons/qgis_icon.svg qgis.svg
+
+	insinto /usr/share/mime/packages
+	doins debian/qgis.xml
+
+	if use examples; then
+		insinto /usr/share/doc/${PF}/examples
+		doins -r "${WORKDIR}"/qgis_sample_data/*
+	fi
+
+	python_optimize "${D}"/usr/share/qgis/python \
```

Comment:
It would be nice. But I think quite a few things to check to do it properly e.g.:
* Can they be replaced by packages from the tree?
* Have qgis modified them?
Unfortunately I haven't got the time at the moment and qgis isn't the only offender in the tree.